### PR TITLE
Use MaybeUninit correctly

### PR DESCRIPTION
**Pull request by haraldh**
`assume_init()` must only be called, when ioctl returns with `0`, otherwise the code path is unsound.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -278,13 +278,15 @@ impl<T> Ioctl<Read, &T> {
     /// internal `ioctl()` call. Upon success, returns the raw (positive)
     /// return value and the instance of `T`.
     pub fn ioctl(self, fd: &impl AsRawFd) -> Result<(c_uint, T)> {
-        let mut out: T = unsafe { core::mem::MaybeUninit::zeroed().assume_init() };
+        use core::mem::MaybeUninit;
```
Comment (npmccallum) [inline src/lib.rs:281]: Move this up top. No reason to bury it in the function.

File: src/lib.rs [hunk 2]
```diff
@@ -278,13 +278,15 @@ impl<T> Ioctl<Read, &T> {
...
 
-        let r = unsafe { ioctl(fd.as_raw_fd(), self.0, &mut out as *mut _, null::<c_void>()) };
+        let mut out: MaybeUninit<T> = MaybeUninit::zeroed();
```
Comment (npmccallum) [inline src/lib.rs:283]: I know this is my fault, but do we want zeroed or uninit here?

Comment (haraldh) [reply src/lib.rs:283]: Hmm.. uninit() might be fine, because the kernel should not expect anything in there yet.

Comment (connorkuehl) [reply src/lib.rs]: I think zeroed is only legal if everything inside of `T` can accept zero as a value

**Pull request merged**